analyzers/src/SonarAnalyzer.Common/Rules/MethodsShouldNotHaveIdenticalImplementationsBase.cs

```diff
@@ -66,13 +70,29 @@ protected override void Initialize(SonarAnalysisContext context)
                                 messageArgs: GetMethodIdentifier(method).ValueText));
                         }
                     }
-                }, ClassDeclarationSyntaxKind);
-        }
+                }, SyntaxKinds);
 
         protected abstract IEnumerable<TMethodDeclarationSyntax> GetMethodDeclarations(SyntaxNode node);
 
         protected abstract SyntaxToken GetMethodIdentifier(TMethodDeclarationSyntax method);
 
         protected abstract bool AreDuplicates(TMethodDeclarationSyntax firstMethod, TMethodDeclarationSyntax secondMethod);
+
+        protected static bool HaveSameParameters<TSyntax>(SeparatedSyntaxList<TSyntax>? leftParameters, SeparatedSyntaxList<TSyntax>? rightParameters)
+            where TSyntax : SyntaxNode
+        {
+            if (leftParameters == null && rightParameters == null)
+            {
+                return true;
+            }
+
+            if (leftParameters == null || rightParameters == null || leftParameters.Value.Count != rightParameters.Value.Count)
+            {
+                return false;
+            }
+
+            return leftParameters.Value.Zip(rightParameters.Value, (p1, p2) => new { p1, p2 })
+                                 .All(tuple => tuple.p1.IsEquivalentTo(tuple.p2, false));
```

Comment:
The  second argument is redundant.

Reply:
About redundancy - same thing as in my first comment